Warning log

mimshwright opened the pull request:
I've added a global object in backend/common that allows you to track errors when doing the conversion from Figma nodes to code. 
The file exposes `warnings` (a `Set<string>`), `addWarning()`, and `clearWarnings()`
The warnings are passed on to the UI to display to the user.
The warnings are also logged to console using `console.warn()`

<img width="854" alt="Screenshot 2024-12-18 at 23 43 57" src="https://github.com/user-attachments/assets/93a49798-2e65-4b69-b245-08a4cef674dc" />

**Known issue:** Any warnings for HTML always appear because they are triggered by the HTML Preview.
**Known issue:** I added warnings for a few conversion issues I was aware of but I assume there are many more that are not annotated.

mimshwright commented:
I also removed some of the extra UI when there were no layers selected. 

<img width="520" alt="image" src="https://github.com/user-attachments/assets/41316902-f654-4260-9451-e63872ed58d3" />

bernaferrari commented on packages/plugin-ui/src/PluginUI.tsx:
```diff
@@ -90,6 +95,18 @@ export const PluginUI = (props: PluginUIProps) => {
           </button>
         </div> */}
+          {warnings.length > 0 && (
+            <div className="flex flex-col bg-red-700 text-white p-4 w-full">
+              <h3 className="text-lg">Warnings:</h3>
+              <ul className="list-disc list-inside">
```
needs to support both light and dark mode, I would do something like bg-red-300 text-black dark:bg-red-700 dark:text-white, or use opacity like dark:bg-red-700/40

bernaferrari replied:
I would probably also use a lucide warning icon and maybe yellow color since it is not THAT bad, just a warning, not an error

bernaferrari commented:
I like it, thanks

mimshwright commented:
I've added the icon and changed the color to yellow. The dark one is slightly darker but changing it too much didn't look correct to me.
Also cleaned up alignment some.

<img width="833" alt="image" src="https://github.com/user-attachments/assets/4a3ef1bf-d223-4342-baf1-4f38246b8184" />

bernaferrari commented:
Oh no, now this has conflicts. Do prefer me to revert the other one, merge this, then merge the other one or do you want to rebase this? Whatever is easier for you. I forgot to merge this, sorry.

mimshwright commented:
This should be rebased and all fixed.

bernaferrari commented:
Thanks!!
